### PR TITLE
[python] add default formatter

### PR DESCRIPTION
**Pull request by hellodword**

### CODE REVIEW

File: src/python/devcontainer-feature.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "id": "python",
-  "version": "1.4.1",
+  "version": "1.4.2",
```
Comment (samruddhikhandale) [inline src/python/devcontainer-feature.json:3]: Can we bump the minor version as we are adding in the new extension?

File: src/python/devcontainer-feature.json [hunk 2]
```diff
@@ -73,15 +73,19 @@
     "vscode": {
       "extensions": [
         "ms-python.python",
-        "ms-python.vscode-pylance"
+        "ms-python.vscode-pylance",
+        "ms-python.autopep8"
```
Comment (samruddhikhandale) [inline src/python/devcontainer-feature.json:77]: This sounds good to me, we always supported it when it was part of `ms-python.python`. So these changes make sense to me. 

// cc @bamurtaugh unless there are any concerns.

**Review by samruddhikhandale — APPROVED** [review]
Looks good to me!

**Comment by hoechenberger** [comment]
Why would you want to use this instead of the Ruff extension?

**Review by samruddhikhandale — APPROVED** [review]
Looks like we missed merging this, approving again ⚡

**Comment by pickfire** [comment]
This broke ruff installation that I had to remove this feature altogether, can this be added behind a feature gate disabled by default? Given that there are quite a few well known alternative, I don't think autopep8 should be enabled by default.

**Comment by hoechenberger** [comment]
@pickfire I added `“-ms-python.autopep8“` to the extensions sections of my dev container config (mind the minus sign, which will remove the extension). And I added a formatted entry to use ruff.

I have to agree that this PR made life harder. Not a fan.

**Comment by samruddhikhandale** [comment]
Okay, revert makes sense in this case. Thanks for bringing this us, feel free to send out a revert PR.

**Pull request merged**